apps/prairielearn/src/models/course-instances.ts

```diff
@@ -0,0 +1,52 @@
+import { callValidatedOneRow } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { IdSchema } from '../lib/db-types';
+import { idsEqual } from '../lib/id';
+
+const CourseInstanceAuthzSchema = z.object({
+  course_instances: z.array(
+    z.object({
+      short_name: z.string().nullable(),
+      long_name: z.string().nullable(),
+      id: IdSchema,
+      formatted_start_date: z.string(),
+      formatted_end_date: z.string(),
+      has_course_instance_permission_view: z.boolean(),
+    }),
+  ),
+});
+
+export async function selectAuthorizedCourseInstancesForCourse({
```

Comment:
As with my other comment, maybe this should be named `selectCourseInstancesWithStaffAccess`?

Reply:
Yep, or maybe `selectStaffCourseInstances()`.

Reply:
I renamed this function in b3e0ec2.